SRE-4803 Upgrade CDN fastly module to use product_enablement feature (#12)
magnolia-sre/terraform-fastly-cdn-service: README.md

| auto_loadbalance   | Indicates whether backend should be included in the pool of backends that requests are load balanced against | bool | false | yes |
| max_connections    | Max connection per backend | number |1000 | yes |
| override_host      | The hostname to override the Host header | string | null | no |

> | shield             | Fastly Point of Presence (POP). [Ref: shielding](https://developer.fastly.com/learning/concepts/shielding/#choosing-a-shield-location). Required whether we are to use it for the fastly `image optimizer` feature for image variations. Whether `shield` parameter is set, `product_enablement` list in its key `image_optimizer` must be set to `true` | string | null | no |

Reviewer: Grammar nitpicking:
```suggestion
| shield             | Fastly Point of Presence (POP). [Ref: shielding](https://developer.fastly.com/learning/concepts/shielding/#choosing-a-shield-location). Required if the `image optimizer` feature for image variations is used. If the `shield` parameter is set, the `image_optimizer` flag has to be included in the `product_enablement` list and be set to `true` | string | null | no |
```

> | product_enablement | Fastly service additional settings configurations. Whether it is set to `true`, the parameter `shield` POP is required too | list | [], thus Fastly defaults | no |

Reviewer: ```suggestion
| product_enablement | Additional fastly features that should be enabled. If `image_optimizer` is included in this list,  the parameter `shield` is required too | list | [], thus Fastly defaults | no |
```
